internal/databases/greenplum/restore_point.go

```diff
@@ -248,6 +248,53 @@ func FindRestorePointBeforeTS(timestampStr string, folder storage.Folder) (strin
 	return targetPoint.Name, nil
 }
 
+// Finds restore point that contains timestamp
+func FindRestorePointWithTS(timestampStr string, folder storage.Folder) (string, error) {
+	ts, err := time.Parse(time.RFC3339, timestampStr)
+	if err != nil {
+		return "", fmt.Errorf("timestamp parse error: %v", err)
+	}
+	// add second because we round down when formatting
+	ts = ts.Add(time.Second)
+
+	restorePointTimes, err := GetRestorePoints(folder.GetSubFolder(utility.BaseBackupPath))
+	if err != nil {
+		return "", err
+	}
+
+	restorePointMetas := make([]RestorePointMetadata, 0)
+	for _, rp := range restorePointTimes {
+		meta, err := FetchRestorePointMetadata(folder, rp.Name)
+		if err != nil {
+			return "", fmt.Errorf("fetch restore point %s metadata: %v", rp.Name, err)
+		}
+
+		restorePointMetas = append(restorePointMetas, meta)
+	}
+
+	var targetPoint *RestorePointMetadata
+	for i := range restorePointMetas {
+		meta := restorePointMetas[i]
+		// target restore point should be started before or right at the provided ts
+		if meta.FinishTime.Before(ts) {
```

Comment:
If restore point should be started before, don't we need to use `meta.StartTime` here?

Reply:
If we indeed need to use `meta.FinishTime`, maybe the comment needs to be changed